src/b-splines/b-splines.jl

```diff
@@ -73,9 +74,16 @@ function interpolate!{IT<:DimSpec{BSpline},GT<:DimSpec{GridType}}(A::AbstractArr
     interpolate!(tweight(A), A, it, gt)
 end
 
+immutable Cubic{BC<:Flag} <: Degree{3} end
+Cubic{BC<:Flag}(::BC) = Cubic{BC}()
+
+immutable Quadratic{BC<:Flag} <: Degree{2} end
+Quadratic{BC<:Flag}(::BC) = Quadratic{BC}()
```

Comment:
They moved here so that I could use `Cubic` and `Quadratic` when re-using some boundary condition implementations across the two degrees. 

If I didn't do this and in `quadratic.jl` I referenced `Cubic`, then I would get an undefined error.

However, this won't be necessary if we re-use implementations from `quadratic.jl` by explicitly forwarding them within `cubic.jl` as suggested above